### ISSUE TITLE
Alternating type, json-schema tuple

### ISSUE THREAD

**Issue by HIAB-vesteel** [ngx-formly/ngx-formly#1627]
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[x] support request
```

Tried to search for closed issues but could not find any matching my thought.
I have a repeating schema of multiple types. 

The first item in the list must be of type A, the second item is of Type B and all other items is of type C.

Json-schema would be
```
{
  "type": array,
  "items": [
    { "type":  A },
    { "type": B }
  ],
  additionalItems: {
    { "type": C }
  }
}
```
I found a similar example on [A list of fixed items](https://ngx-formly.github.io/ngx-formly/examples/advanced/json-schema). By the way, in json-schema its called a tuple.

**Comment by kenisteward** [comment]
Could you give an example of what you'd want in more detail? The example
you sent seems to work just fine for the numbers case.

In that case type 1 is string type 2 is Boolean and the additional items
type is number and it gives more numbers as you click add.

Do you mean referenced types?

**Comment by HIAB-vesteel** [comment]
The example i pointed to is using your `formlyJsonschema.toFieldConfig` to create the fields. I would like to build the same feature but without JSON-schemas. And i was looking for a small code example of how it could be done.

When thinking of how to solve the same situation i came up with 
`    const config: FormlyFieldConfig[] = [
      {
        type: 'repeat',
        templateOptions: {
          addText: 'Add',
          label: 'Items'
        },
        // fieldArray dosen´t accept an array so we need a fieldGroup
        fieldArray: {
          fieldGroup: [
            {
              type: 'input',
              templateOptions: {
                label: 'type A'
              }
            },
            {
              type: 'input',
              templateOptions: {
                label: 'type B'
              }
            },
            {
              type: 'input',
              templateOptions: {
                label: 'type C'
              }
            }
          ],
        }
      }
    ];`

It feels like the key to getting this to work i to disable and hide type B and type C for index 0.
Disable and hide type A and type C for index 1.
And for index is 1 or greater disable and hide type A and type B.

This feels incredibly backwards and must be slowish.  And also when setting data which type will actually get the data. As i have tried it type C always gets the data. 

Side note, type A, type B and type C is objects with different fields (shared base). 

Basically my question is, am i barking up the wrong tree? How does  `formlyJsonschema.toFieldConfig` solve the same issue.

**Comment by kenisteward** [comment]
My advice would be for you to create a simple test schema or open the
example in stackblitz. The toFormlyConfig function returns the full config
to be bound to an ngx-formly form

That will show you the configuration needed for what you are trying to
achieve.

**Comment by HIAB-vesteel** [comment]
I don't know why i didn't pull the schema down from toFormlyConfig and check it. Thanks for the tip.

I get some more follow up questions... what is this.  This looks like a home hack and not a permanent solution. 
We have a fieldGroup containing a fieldArray of type string, but also a field group which is empty.
Then some one added non standard keys _fieldArray and _additionalFieldArray. The interface  FormlyFieldConfig does not allow these keys and gives error.. 

I guess you have an internal extended FormleyFieldConfig with the new parameters and a whole new internal view/code to make this work which i have to mess around in the source code to get a sense of how it work.

Once again i feel like, this can't be the intended use... ? Am i trying to do something too complex for ngx-formley? It feels like arrays in general in ngx-formley have become way to complicated for everyday use. 

Any pointers, help or information would be deeply appreciated.


```
[
     {
		"type": "object",
		"templateOptions": {},
		"fieldGroup": [{
				"key": "fixedItemsList",
				"type": "array",
				"templateOptions": {
					"label": "A list of fixed items"
				},
				"_fieldArray": [{
						"key": "fixedItemsList",
						"type": "string",
						"defaultValue": "lorem ipsum",
						"templateOptions": {
							"label": "A string value"
						}
					}, {
						"key": "fixedItemsList",
						"type": "boolean",
						"templateOptions": {
							"label": "a boolean value"
						}
					}
				],
				"fieldGroup": [],
				"_additionalFieldArray": {
					"key": "fixedItemsList",
					"type": "number",
					"templateOptions": {
						"label": "Additional item"
					}
				},
				"fieldArray": {
					"key": "fixedItemsList",
					"type": "string",
					"defaultValue": "lorem ipsum",
					"templateOptions": {
						"label": "A string value"
					}
				}
			}
		]
	}
]

```

**Comment by aitboudad** [comment]
we rely on a getter to achieve that not ready yet for formly json as we're still experimenting with the `json-schema` and once done we'll try to make formly json much more powerful.

Right now we achieved that using a getter property, I'll provide you a workaround next week.

**Comment by HIAB-vesteel** [comment]
Thank you aitboudad!

I like to hear that you are looking in to support `json-schema`.  I/we are actually using `json-schema` right now but rewrite it our self to ngx-formley configs as the support is not good enough today.

**Comment by aitboudad** [comment]
here is an example using formly json config: https://stackblitz.com/edit/angular-rhagg9
